### PR TITLE
Added Nerdicons Plugin

### PR DESCRIPTION
**Pull request by techie-guy**
# Nerdicons Plugin

### **This is an extension/modification of the [nonicons plugins](https://github.com/lite-xl/lite-xl-plugins/blob/master/plugins/nonicons.lua)**

**Any icon can be searched by it's hex from [nerdfonts.com/cheat-sheet](https://nerdfonts.com/cheat-sheet)**

## How to use:
1. Download a nerd font that you like from [nerdfonts.com](https://nerdfonts.com)
2. Unzip the zip file that you downloaded.
3. Choose a font file from the unzipped directory and install it.
4. Copy the font file to the lite-xl data directory and rename it to "icon-nerd-font.ttf".
5. Restart lite-xl, the icons should appear.

### CODE REVIEW

File: plugins/nerdicons.lua [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+-- This is an extension/modification of the nonicons plugin, https://github.com/lite-xl/lite-xl-plugins/blob/master/plugins/nonicons.lua
+-- If you have any issues or want to suggest any changes, please create an issue and add [nerdicons] in the title
+
+-- Any icon can be searched by it's hex from nerdfonts.com/cheat-sheet
+
+-- How to use:
+-- 1. Download a nerd font that you like from nerdfonts.com
+-- 2. Unzip the zip file that you downloaded.
+-- 3. Choose a font file from the unzipped directory and install it.
+-- 4. Copy the font file to the lite-xl-dir/data/fonts and rename it to "icon-nerd-font.ttf".
```
Comment (Guldoman) [inline plugins/nerdicons.lua:15]: Don't use the data dir for user-installed things. Use the user dir (.config/lite-xl).

File: plugins/nerdicons.lua [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      label = "Draw Tab Icons",
+      description = "Adds file related icons to tabs.",
+      path = "draw_tab_icons",
+      type = "toggle",
+      default = true
+    },
+  }
+}, config.plugins.nerdicons)
+
+local icon_font = renderer.font.load(DATADIR .. "/fonts/icon-nerd-font.ttf", 18.5 * SCALE)
```
Comment (Guldoman) [inline plugins/nerdicons.lua:65]:
```suggestion
local icon_font = renderer.font.load(USERDIR .. "/fonts/icon-nerd-font.ttf", 18.5 * SCALE)
```

**Review by Guldoman — CHANGES_REQUESTED** [review]
I'll review more in depth later.

**Comment by jgmdev** [comment]
Whoa! Nice icons collection, seems to have all I need  for some plugins.

**Comment by adamharrison** [comment]
Hey @techie-guy; I'm doing a big push to merge PRs and stuff; can you update this one to use the new manifest system instead? (Sorry). I'll review it as soon as that's done.

**Pull request closed without merging**